lxd-agent/api_1.0.go

```diff
@@ -61,3 +66,57 @@ func api10Get(d *Daemon, r *http.Request) response.Response {
 
 	return response.SyncResponseETag(true, fullSrv, fullSrv)
 }
+
+func api10Put(d *Daemon, r *http.Request) response.Response {
+	var data API10Put
+
+	err := json.NewDecoder(r.Body).Decode(&data)
+	if err != nil {
+		return response.ErrorResponse(http.StatusInternalServerError, err.Error())
+	}
+
+	d.serverCID = data.CID
+	d.serverPort = data.Port
+
+	// Try connecting to LXD server.
+	client, err := getClient(int(d.serverCID), int(d.serverPort))
+	if err != nil {
+		return response.ErrorResponse(http.StatusInternalServerError, err.Error())
+	}
+
+	server, err := lxd.ConnectLXDHTTP(nil, client)
+	if err != nil {
+		return response.ErrorResponse(http.StatusInternalServerError, err.Error())
+	}
+
+	defer server.Disconnect()
+
+	// Let LXD know, we were able to connect successfully.
+	d.chConnected <- struct{}{}
+
+	return response.EmptySyncResponse
+}
+
+func getClient(CID int, port int) (*http.Client, error) {
+	agentCert, err := ioutil.ReadFile("/run/lxd_agent/agent.crt")
+	if err != nil {
+		return nil, err
+	}
+
+	agentKey, err := ioutil.ReadFile("/run/lxd_agent/agent.key")
+	if err != nil {
+		return nil, err
+	}
+
+	serverCert, err := ioutil.ReadFile("/run/lxd_agent/lxd.crt")
+	if err != nil {
+		return nil, err
+	}
```

Comment:
We shouldn't be hardcoding the whole path but instead just assume that we are in the correct directory and load agent.crt, agent.key and lxd.crt from there.

Also, how's `lxd.crt` different from `server.crt`?